src/main/java/fr/pickaria/pterodactylpoweraction/ServerStartBossBar.java

```diff
@@ -0,0 +1,191 @@
+            new Phase(Pattern.compile("Preparing (?:level|start region|spawn area|world data)"), "bossbar.preparing.world", 0.90f),
+            new Phase(Pattern.compile("Done \\([0-9.]+s\\)!|Listening on"), "bossbar.done", 1.0f)
+    );
+
+    ServerStartBossBar(StartingServer audience, Configuration configuration, String serverName, Logger logger) {
+        this.audience = audience;
+        this.configuration = configuration;
+        this.serverName = serverName;
+        this.logger = logger;
+        this.bossBar = BossBar.bossBar(
+                Component.translatable(messageKey, Component.text(serverName)),
+                0f,
+                BossBar.Color.YELLOW,
+                BossBar.Overlay.PROGRESS
+        );
+    }
+
+    void start() {
+        messageKey = "bossbar.starting";
+        dots = 0;
+        bossBar.progress(0f);
+        bossBar.name(Component.translatable(messageKey, Component.text(serverName)));
+        audience.showBossBar(bossBar);
+        scheduler = Executors.newSingleThreadScheduledExecutor();
+        scheduler.scheduleAtFixedRate(this::animate, 1, 1, TimeUnit.SECONDS);
+        scheduler.schedule(() -> messageKey = "bossbar.taking.long", MAX_DURATION.toSeconds(), TimeUnit.SECONDS);
+        scheduler.schedule(() -> {
+            if (webSocket != null) webSocket.abort();
+        }, MAX_DURATION.toSeconds(), TimeUnit.SECONDS);
+        if (configuration.getAPIType() == APIType.PTERODACTYL) {
+            startWebSocketWatcher();
+        }
+    }
+
+    void addPlayer(Player player) {
+        player.showBossBar(bossBar);
+    }
+
+    void stop() {
+        if (webSocket != null) {
+            webSocket.abort();
+        }
+        if (scheduler != null) {
+            scheduler.shutdownNow();
+        }
+        audience.hideBossBar(bossBar);
+        messageKey = "bossbar.starting";
+        dots = 0;
+        bossBar.progress(0f);
+        bossBar.name(Component.translatable(messageKey, Component.text(serverName)));
+    }
+
+    void removePlayer(Player player) {
+        player.hideBossBar(bossBar);
+    }
+
+    private void animate() {
+        dots = (dots % 3) + 1;
+        String dotsStr = ".".repeat(dots);
+        Component base;
+        if ("bossbar.starting".equals(messageKey)) {
+            base = Component.translatable(messageKey, Component.text(serverName));
+        } else {
+            base = Component.translatable(messageKey);
+        }
+        bossBar.name(base.append(Component.text(dotsStr)));
+    }
+
+    private void handleConsole(String line) {
+        for (Phase phase : PHASES) {
+            if (phase.pattern.matcher(line).find()) {
+                messageKey = phase.key;
+                bossBar.progress(Math.max(bossBar.progress(), phase.progress));
+                break;
+            }
+        }
+    }
+
+    private void startWebSocketWatcher() {
```

Comment:
Anything specific to one API must be in the appropriate API class. When we will implement support for [Pelican](https://pelican.dev/), I don't want to go around and update code all over the place.
Also, I believe the websocket for Pterodactyl is already implemented somewhere else, it could be wise to refactor the code to re-use the websocket logic instead of re-writing it every time.